hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/DFSUtil.java

```diff
@@ -1936,4 +1936,20 @@ public static boolean isParentEntry(final String path, final String parent) {
     return path.charAt(parent.length()) == Path.SEPARATOR_CHAR
         || parent.equals(Path.SEPARATOR);
   }
+
+  /**
+   * Calculate the transfer rate in megabytes/second. Return -1 for any negative input.
+   * @param bytes bytes
+   * @param durationMS duration in milliseconds
+   * @return the number of megabytes/second of the transfer rate
+  */
+  public static long transferRateMBs(long bytes, long durationMS) {
+    if (bytes < 0 || durationMS < 0) {
+      return -1;
+    }
+    if (durationMS == 0) {
+      durationMS = 1;
+    }
+    return bytes / (1024 * 1024) * 1000 / durationMS;
```

Comment:
We probably need to change the division to be done based on float/double. Then return a float/double as well.

Reply:
The Metrics take long, so eventually we will have to convert to long anyways.

Reply:
I would make this bytes per second and let the consumer do the conversion.

Reply:
If we want to have the consumer do the conversion, then we can just keep it to bytes/milliseconds. bytes and milliseconds are the units used for most of the metrics in HDFS.
Thoughts @xinglin @mkuchenbecker ?